text/0000-fabric-gateway.md

```diff
@@ -0,0 +1,213 @@
+---
+layout: default
+title: Fabric Gateway
+nav_order: 3
+---
+
+- Feature Name: Fabric Gateway
+- Start Date: 2020-09-08
+- RFC PR: (leave this empty)
+- Fabric Component: fabric-gateway
+- Fabric Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The Fabric Gateway is a new component that will run either as a standalone process, or embedded with the peer, and will implement as much of the high-level 'gateway' programming model as possible.  This will remove much of the transaction submission logic from the client application which will simplify the maintenance of the SDKs.  It will also simplify the administrative overhead of running a Fabric network because client applications will be able to connect and submit transactions via a single network port rather than the current situation where ports have to be opened to multiple peers across potentially multiple organisations.
+
+# Motivation
+[motivation]: #motivation
+
+The high-level "gateway" programming model has proved to be very popular with users since its introduction in v1.4 in Node SDK.  Since then, it has been implemented in the Java and Go SDKs.  This has led to a large amount of code that has to be maintained in the SDKs which increases the cost of implementing new feature which have to be replicated to keep the SDKs in step.  It also increases the cost of creating an SDK for a new language.  Moving all of the high-level logic into its own process (or peer) and exposing a simplified gRPC interface will drastically reduce the complexity of the SDKs.
```

Comment:
It means if this RFC passed, stop maintaining work of the high-level "gateway" programming model will be on schedule?

Reply:
> In addition to the comments on the text:
> 
> * Will there be health checks and metrics?

They both seem like nice things to have.

> * What sort of authentication and authorization will be implemented on the service?

https://github.com/hyperledger/fabric-rfcs/pull/32#discussion_r488820919

Reply:
I think the intention is for a Fabric Gateway client SDK to provide pretty much the same API as the existing high-level Fabric programming model. Moving as much of the implementation as practical out of the client-side SDK implementation into a common Fabric Gateway component just helps to reduce the amount of effort required to maintain the "gateway" programming model across multiple client-side languages.